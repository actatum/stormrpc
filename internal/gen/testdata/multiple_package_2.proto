syntax = "proto3";

package test;

option go_package = "./food;food";

service Food {
    rpc SayFood (FoodRequest) returns (FoodReply) {}
}

message FoodRequest {
    string name = 1;
}

message FoodReply {
    string message = 1;
}
