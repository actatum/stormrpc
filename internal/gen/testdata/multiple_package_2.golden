// Code generated by protoc-gen-stormrpc. DO NOT EDIT.

package food

import (
	context "context"
	fmt "fmt"
	stormrpc "github.com/actatum/stormrpc"
)

// FoodClient is the client API for Food service.
type FoodClient interface {
	SayFood(ctx context.Context, in *FoodRequest, opts ...stormrpc.CallOption) (*FoodReply, error)
}

type foodClient struct {
	c *stormrpc.Client
}

func NewFoodClient(c *stormrpc.Client) FoodClient {
	return &foodClient{c}
}

func (c *foodClient) SayFood(ctx context.Context, in *FoodRequest, opts ...stormrpc.CallOption) (*FoodReply, error) {
	var out FoodReply
	r, err := stormrpc.NewRequest("rpc.Food.SayFood", in, stormrpc.WithEncodeProto())
	if err != nil {
		return nil, err
	}

	resp := c.c.Do(ctx, r, opts...)
	if resp.Err != nil {
		return nil, resp.Err
	}

	if err = resp.Decode(&out); err != nil {
		return nil, err
	}

	return &out, nil
}

// FoodServer is the server API for Food service.
type FoodServer interface {
	SayFood(context.Context, *FoodRequest) (*FoodReply, error)
}

func RegisterFoodServer(s *stormrpc.Server, srv FoodServer) {
	for _, handler := range foodHandlers {
		handler.SetService(srv)
		s.Handle(handler.Route(), handler.HandlerFunc())
	}
}

type _Food_SayFood_Handler struct {
	route string
	svc   interface{}
}

func (h *_Food_SayFood_Handler) HandlerFunc() stormrpc.HandlerFunc {
	return func(ctx context.Context, r stormrpc.Request) stormrpc.Response {
		var in FoodRequest
		if err := r.Decode(&in); err != nil {
			return stormrpc.NewErrorResponse(r.Reply, fmt.Errorf("error decoding request"))
		}

		out, err := h.svc.(FoodServer).SayFood(ctx, &in)
		if err != nil {
			return stormrpc.NewErrorResponse(r.Reply, err)
		}

		resp, err := stormrpc.NewResponse(r.Reply, out, stormrpc.WithEncodeProto())
		if err != nil {
			return stormrpc.NewErrorResponse(r.Reply, err)
		}

		return resp
	}
}

func (h *_Food_SayFood_Handler) Route() string {
	return h.route
}

func (h *_Food_SayFood_Handler) SetService(svc interface{}) {
	h.svc = svc
}

var foodHandlers = []handler{
	&_Food_SayFood_Handler{route: "rpc.Food.SayFood"},
}

type handler interface {
	Route() string
	HandlerFunc() stormrpc.HandlerFunc
	SetService(interface{})
}
